client-datastore-inmemory/src/main/java/com/telecominfraproject/wlan/client/datastore/inmemory/ClientDatastoreInMemory.java

```diff
@@ -176,6 +176,120 @@ public Client delete(int customerId, MacAddress clientMac) {
         return ret;
     
     }
+    
+    @Override
+    public PaginationResponse<Client> searchByMacAddress(int customerId, String macSubstring,
+    		final List<ColumnAndSort> sortBy, PaginationContext<Client> context) {
+    	
+    	if(context == null) {
+    		context = new PaginationContext<>();
+    	}
+
+        PaginationResponse<Client> ret = new PaginationResponse<>();
+        ret.setContext(context.clone());
+
+        if (ret.getContext().isLastPage()) {
+            // no more pages available according to the context
+            return ret;
+        }
+
+        List<Client> items = new LinkedList<>();
+        
+    	if(macSubstring != null) {
```

Comment:
This is the only null check for the macSubstring in the whole PR.
Usually null or empty criteria means that caller wants to see all items.
The way you implemented it right now for the null criteria - InMemory datastore will return no items, RDBMS and Cassandra datastores will throw NullPointerException.
Please align the datastores on common behaviour (I suggest to return all items), and add unit tests for the null and "" macSubstring criteria.